libs/features/mas/src/aem-fragment.js

```diff
@@ -147,22 +102,65 @@ export class AemFragment extends HTMLElement {
     }
 
     connectedCallback() {
+        this.#service = getService(this);
+        this.#log = this.#service.log.module(AEM_FRAGMENT_TAG_NAME);
+        this.#startMark = `${AEM_FRAGMENT_TAG_NAME}:${this.#fragmentId}${MARK_START_SUFFIX}`;
+        performance.mark(this.#startMark);
         if (!this.#fragmentId) {
             this.#fail({ message: 'Missing fragment id' });
             return;
         }
-        this.#startMark = `${AEM_FRAGMENT_TAG_NAME}:${this.#fragmentId}${MARK_START_SUFFIX}`;
-        performance.mark(this.#startMark);
-        this.#readyPromise = new Promise((resolve, reject) => {
-          this.dispose = discoverService((masCommerceService) => this.activate(masCommerceService, resolve, reject));
-        });
+        this.activate();
+    }
+
+    /**
+     * Get fragment by ID
+     * @param {string} endpoint url to fetch fragment from
+     * @param {string} id fragment id
+     * @returns {Promise<Object>} the raw fragment item
+     */
+    async getFragmentById(endpoint, id, startMark) {
+        const measureName = `${AEM_FRAGMENT_TAG_NAME}:${id}${MARK_DURATION_SUFFIX}`;
+        let response;
+        try {
+            response = await masFetch(endpoint, {
+                cache: 'default',
+                credentials: 'omit',
+            });
+            if (!response?.ok) {
+                const { startTime, duration } = performance.measure(
+                    measureName,
+                    startMark,
+                );
+                throw new MasError('Unexpected fragment response', {
+                    response,
+                    startTime,
+                    duration,
+                    ...this.#service.duration,
+                });
+            }
+            return response.json();
+        } catch (e) {
+            const { startTime, duration } = performance.measure(
+                measureName,
+                startMark,
+            );
+            if (!response) {
+                response = { url: endpoint };
+            }
+
+            throw new MasError('Failed to fetch fragment', {
+                response,
+                startTime,
+                duration,
+                ...this.#service.duration,
+            });
+        }
     }
 
-    async activate(masCommerceService, resolve, reject) {
-        this.#service = masCommerceService;
-        this.refresh(false)
-            .then((result) => resolve(result))
-            .catch((e)=> reject(e));
+    async activate() {
+        const flushCache = !this.#author;
```

Comment:
i remember there was the reason why we changed it just a week ago in Angelo's PR
https://github.com/adobecom/milo/pull/3786/files#diff-36078108e571d0861359f4696fe62484ed743f80e52d1e082c7c37b858b08955R163

did you had to bring it back because something wasn't working?

Reply:
might be due to merging, I'll double check thanks.